openhtf/output/web_gui/__init__.py

```diff
@@ -441,23 +455,19 @@ def handle_test_state_update(self, hostport, test_uid, state):
     if state is None:
       return
 
-    def make_plug_url(plug_name):
-      """Plug URL is /plugs/<host>/<port>/<test_uid>/<plug_name>"""
-      return '/plugs/%s/%s/%s/%s' % (hostport + (test_uid, plug_name))
-
     StationPubSub.publish_test_state_update(hostport, test_uid, state)
     plugs_port = state.plugs and state.plugs['xmlrpc_port']
-    if plugs_port:
-      # See plugs.RemotePlug.discover for details. This call yields tuples of
-      # (plug_name, handler, handler_params). From the plug names, we create
-      # URLs that are particular to a given station and test_uid.
-      handlers = [
-          (make_plug_url(plug_name), handler, params)
-          for plug_name, handler, params
-          in plugs.RemotePlug.discover(hostport.host, plugs_port)
-          if not self.has_handler_for_url(make_plug_url(plug_name))
-      ]
-      self.add_handlers('.*$', handlers)
+    if not plugs_port:
+      return
+
+    # Update the plugs we're interacting with. The PlugManager we're interacting
+    # with may have gone and come back since the last update, so we have to
+    # open a new connection each time to avoid getting an update
```

Comment:
Update the plugs we're interacting with. On each state update, we open new connections to the XML-RPC server via RemotePlug.discover because the PlugManager we're interacting with may have gone and come back since the last update.